src/delphi/eval/token_labelling.py

```diff
@@ -0,0 +1,211 @@
+        print("Token POS:".ljust(20), spacy.glossary.explain(token.pos_))
+        print(" Token labels ".center(45, "-"))
+        for i, (label_name, value) in enumerate(labels.items()):
+            print(f" {i:2}  ", label_name.ljust(20), value)
+
+    else:
+        glossary = spacy.glossary.GLOSSARY
+        print(
+            f"Explanation of all {len(glossary.keys())} token labels (POS, dependency, NER, ...):"
+        )
+        for label, key in glossary.items():
+            print("   ", label.ljust(10), key)
+
+
+def label_single_Token(token: Token) -> dict[str, bool]:
+    """
+    Labels a single token. A token, that has been analyzed by the spaCy
+    library.
+
+    Parameters
+    ----------
+    token : Token
+        The token to be labelled.
+
+    Returns
+    -------
+    dict[str, bool]
+        Returns a dictionary with the token's labels as keys and their
+        corresponding boolean values.
+    """
+    assert isinstance(token, Token)
+    labels = dict()  #  The list holding labels of a single token
+    for label_name, category_check in TOKEN_LABELS.items():
+        labels[label_name] = category_check(token)
+    return labels
+
+
+def label_sentence(tokens: Union[Doc, list[Token]]) -> list[dict[str, bool]]:
+    """
+    Labels spaCy Tokens in a sentence. Takes the context of the token into account
+    for dependency labels (e.g. subject, object, ...), IF dependency labels are turned on.
+
+    Parameters
+    ----------
+    tokens : list[Token]
+        A list of tokens.
+
+    Returns
+    -------
+    list[dict[str, bool]]
+        Returns a list of the tokens' labels.
+    """
+    labelled_tokens = list()  # list holding labels for all tokens of sentence
+    for token in tokens:
+        labels = label_single_Token(token)
+        labelled_tokens.append(labels)
+    return labelled_tokens
+
+
+def label_batch_sentences(
+    sentences: Union[list[str], list[list[str]]],
+    tokenized: bool = True,
+    verbose: bool = False,
+) -> list[list]:
+    """
+    Labels tokens in a sentence batchwise. Takes the context of the token into
+    account for dependency labels (e.g. subject, object, ...).
+
+    Parameters
+    ----------
+    sentences : list
+        A batch/list of sentences, each being a list of tokens.
+    tokenized : bool, optional
+        Whether the sentences are already tokenized, by default True. If the sentences
+        are full strings and not lists of tokens, then set to False. If true then `sentences` must be list[list[str]].
+    verbose : bool, optional
+        Whether to print the tokens and their labels to the console, by default False.
+
+    Returns
+    -------
+    list[list[dict[str, bool]]
+        Returns a list of sentences. Each sentence contains a list of its
+        corresponding token length where each entry provides the labels/categories
+        for the token. Sentence -> Token -> Labels
+    """
+    assert isinstance(sentences, list)
```

Comment:
Redundant, we're running beartype. Remove it and try passing something different than specified type, it should fail.